pkg/model/context.go

```diff
@@ -153,6 +153,16 @@ func (b *KopsModelContext) CloudTagsForInstanceGroup(ig *kops.InstanceGroup) (ma
 		labels[fi.StringValue(nth.ManagedASGTag)] = ""
 	}
 
+	switch b.Cluster.Spec.GetCloudProvider() {
+	case kops.CloudProviderAWS:
+		// Apply labels for cluster autoscaler autodiscovery
+		cas := b.Cluster.Spec.ClusterAutoscaler
+		if cas != nil && fi.BoolValue(cas.Enabled) && fi.BoolValue(cas.AutoDiscovery) && ig.Spec.Autoscale != nil {
```

Comment:
This belongs in a separate PR I think. Debatable if this feature is needed at all and what value it would bring.

Reply:
Hi @olemarkus cluster-autoscaler autodiscovery have been moved https://github.com/kubernetes/kops/pull/14521